Source/WebCore/html/PermissionsPolicy.cpp

```diff
@@ -39,62 +39,171 @@ namespace WebCore {
+        feature = PermissionsPolicy::Feature::SpeakerSelection;
+        remainingValue = value.substring(speakerSelectionToken.length());
+    } else if (value.startsWith(displayCaptureToken)) {
+        feature = PermissionsPolicy::Feature::DisplayCapture;
+        remainingValue = value.substring(displayCaptureToken.length());
+    } else if (value.startsWith(gamepadToken)) {
+        feature = PermissionsPolicy::Feature::Gamepad;
+        remainingValue = value.substring(gamepadToken.length());
+    } else if (value.startsWith(geolocationToken)) {
+        feature = PermissionsPolicy::Feature::Geolocation;
+        remainingValue = value.substring(geolocationToken.length());
+    } else if (value.startsWith(paymentToken)) {
+        feature = PermissionsPolicy::Feature::Payment;
+        remainingValue = value.substring(paymentToken.length());
+    } else if (value.startsWith(screenWakeLockToken)) {
+        feature = PermissionsPolicy::Feature::ScreenWakeLock;
+        remainingValue = value.substring(screenWakeLockToken.length());
+    } else if (value.startsWith(syncXHRToken)) {
+        feature = PermissionsPolicy::Feature::SyncXHR;
+        remainingValue = value.substring(syncXHRToken.length());
+    } else if (value.startsWith(fullscreenToken)) {
+        feature = PermissionsPolicy::Feature::Fullscreen;
+        remainingValue = value.substring(fullscreenToken.length());
+    } else if (value.startsWith(webShareToken)) {
+        feature = PermissionsPolicy::Feature::WebShare;
+        remainingValue = value.substring(webShareToken.length());
+#if ENABLE(DEVICE_ORIENTATION)
+    } else if (value.startsWith(gyroscopeToken)) {
+        feature = PermissionsPolicy::Feature::Gyroscope;
+        remainingValue = value.substring(gyroscopeToken.length());
+    } else if (value.startsWith(accelerometerToken)) {
+        feature = PermissionsPolicy::Feature::Accelerometer;
+        remainingValue = value.substring(accelerometerToken.length());
+    } else if (value.startsWith(magnetometerToken)) {
+        feature = PermissionsPolicy::Feature::Magnetometer;
+        remainingValue = value.substring(magnetometerToken.length());
+#endif
+#if ENABLE(WEB_AUTHN)
+    } else if (value.startsWith(publickeyCredentialsGetRuleToken)) {
+        feature = PermissionsPolicy::Feature::PublickeyCredentialsGetRule;
+        remainingValue = value.substring(magnetometerToken.length());
+#endif
+#if ENABLE(WEBXR)
+    } else if (value.startsWith(xrSpatialTrackingToken)) {
+        feature = PermissionsPolicy::Feature::XRSpatialTracking;
+        remainingValue = value.substring(xrSpatialTrackingToken.length());
+#endif
+    } else if (value.startsWith(privateTokenToken)) {
+        feature = PermissionsPolicy::Feature::PrivateToken;
+        remainingValue = value.substring(privateTokenToken.length());
+    }
+
+    if (remainingValue.startsWith(":"_s))
+        remainingValue = remainingValue.substring(1);
```

Comment:
Maybe add a FIXME to remove this rule?
If we can do that, we could potentially further align and check for whitespace as a way to split the feature name.

Reply:
Will add.